src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/SliceTriangleGeometry.cpp

```diff
@@ -321,63 +88,51 @@ Result<> SliceTriangleGeometry::operator()()
-    verts[3 * (2 * i + 1)] = slicedVerts[3 * (2 * i + 1)];
-    verts[3 * (2 * i + 1) + 1] = slicedVerts[3 * (2 * i + 1) + 1];
-    verts[3 * (2 * i + 1) + 2] = slicedVerts[3 * (2 * i + 1) + 2];
-    sliceId[i] = sliceIds[i];
+    verts[3 * (2 * i)] = sliceTriangleResult.SliceVerts[3 * (2 * i)];
+    verts[3 * (2 * i) + 1] = sliceTriangleResult.SliceVerts[3 * (2 * i) + 1];
+    verts[3 * (2 * i) + 2] = sliceTriangleResult.SliceVerts[3 * (2 * i) + 2];
+    verts[3 * (2 * i + 1)] = sliceTriangleResult.SliceVerts[3 * (2 * i + 1)];
+    verts[3 * (2 * i + 1) + 1] = sliceTriangleResult.SliceVerts[3 * (2 * i + 1) + 1];
+    verts[3 * (2 * i + 1) + 2] = sliceTriangleResult.SliceVerts[3 * (2 * i + 1) + 2];
+    sliceId[i] = sliceTriangleResult.SliceIds[i];
     if(m_InputValues->HaveRegionIds)
     {
-      (*triRegionIds)[i] = regionIds[i];
+      (*triRegionIds)[i] = sliceTriangleResult.RegionIds[i];
     }
   }
 
-  // rotate all CAD triangles back to original orientation
-  // rotateVertices(rotBackward, n, numTriVerts, triVerts);
-  // rotate all edges back to original orientation
-  // rotateVertices(rotBackward, n, numVerts, verts);
-
-  // m_MessageHandler("Complete");
-
-  Result<> result = GeometryUtilities::EliminateDuplicateNodes<EdgeGeom>(edge);
-  return result;
-}
+  Result<> result = GeometryUtilities::EliminateDuplicateNodes<EdgeGeom>(edgeGeom);
+  if(result.invalid())
+  {
+    return result;
+  }
 
-// -----------------------------------------------------------------------------
-usize SliceTriangleGeometry::determineBoundsAndNumSlices(float32& minDim, float32& maxDim, usize numTris, TriStore& tris, VertsStore& triVerts)
-{
-  for(usize i = 0; i < numTris; i++)
+  // REMOVE DUPLICATE EDGES FROM THE GENERATED EDGE GEOMETRY
+  // Remember to also fix up the sliceIds and regionIds arrays
+  using UniqueEdges = std::set<std::pair<uint64, uint64>>;
+  UniqueEdges uniqueEdges;
+  usize currentEdgeListSize = 0;
+  for(usize edgeIdx = 0; edgeIdx < numEdges; edgeIdx++)
   {
-    for(usize j = 0; j < 3; j++)
+    uniqueEdges.insert(std::minmax(edges[edgeIdx * 2], edges[edgeIdx * 2 + 1]));
+    // Did something get inserted
+    if(currentEdgeListSize < uniqueEdges.size())
     {
-      const usize vert = tris[3 * i + j];
-      if(minDim > triVerts[3 * vert + 2])
-      {
-        minDim = triVerts[3 * vert + 2];
-      }
-      if(maxDim < triVerts[3 * vert + 2])
+      edges[currentEdgeListSize * 2] = edges[edgeIdx * 2];
+      edges[currentEdgeListSize * 2 + 1] = edges[edgeIdx * 2 + 1];
+      sliceId[currentEdgeListSize] = sliceId[edgeIdx];
+      if(m_InputValues->HaveRegionIds)
       {
-        maxDim = triVerts[3 * vert + 2];
+        (*triRegionIds)[currentEdgeListSize] = (*triRegionIds)[edgeIdx];
       }
+      currentEdgeListSize++;
     }
   }
-
-  // adjust sectioning range if user selected a specific range - check that user range is within actual range
-  if(m_InputValues->SliceRange == 1)
+  if(numEdges != uniqueEdges.size())
```

Comment:
Inside this if statement, you resize the edge list, but how can you guarantee the edges being chopped off the array are the duplicates? Are the edges pre-generated/pre-sorted such that the duplicates are on the end? Or is it that the edges aren't actually calculated here and you are just defining lengths?

I considered maybe one of the resize functions did it implicitly, but the `resizeEdgeList()` function doesn't recalculate the edges.

Reply:
This is a bit non-obvious but in the loop above this section starting at line 115 we are effectively rewriting through the shared edges array. If we encounter an edge that has already been written (through the std::set<> variable) then we skip writing that edge which means we also skip incrementing the `currentEdgeListSize` variable.
   
 So when we finally hit that `if` statement on line 131 if the 2 sizes do NOT match we resize the edges array which by the algorithm the new size is always smaller in value than the previous size. The way that DataArray works when you resize, it will always copy what elements it can from the old array into the new array.